### ISSUE TITLE
onKeyDown prop causes field to lose focus on every keystroke

### ISSUE THREAD

**Issue by ashleyw** [ashleyw/react-sane-contenteditable#16]

**Comment by nijk** [comment]
@ashleyw Do you think this is related to the latest change in `v1.3.5`?

**Comment by nijk** [comment]
Ah, I've found the issue. It seems to pre-date `v1.3.5` and it seems to be the `forceUpdate` [on this line](https://github.com/ashleyw/react-sane-contenteditable/blob/7a6931faa04e794ab6fcc349c2579756c8c4f568/src/react-sane-contenteditable.js#L46) that causes it.

A few things that occurred to me whilst I was debugging this:

1. `componentWillReceiveProps` is [marked for deprecation](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops) as of React 16.
2. I'm not sure I understand why forceUpdate is necessary - perhaps there is a fight going on between the props coming into the component and the state being stored within the component?
3. I don't understand why the `content` prop is being set in state, there doesn't seem to be anything in the component that *needs* to have the initial prop set as state (seeding the `content` prop). As a test I removed all `setState` calls in the component and used `dangerouslySetInnerHTML={{ __html: this.sanitiseValue(content) }}` in the render instead and I can't see any difference in behaviour when running the demo. That said I haven't covered all of the use cases robustly by configuring the component in different ways via its other props. Perhaps this is something that good test coverage would help with.
